### PR TITLE
Remove deprecations in convolution

### PR DESCRIPTION
**Pull request by bsipocz**
This is to address part of #8888 

I'll also go ahead and push some more cleanup like fixes to this branch I made as my editor was like a Christmas tree. Happy to refactor those out, but I feel any cleanup would logically fit in here.

### CODE REVIEW

File: astropy/convolution/convolve.py [hunk 1]
```diff
@@ -374,14 +375,12 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
         return result
 
 
-@deprecated_renamed_argument('interpolate_nan', 'nan_treatment', 'v2.0.0')
 @support_nddata(data='array')
 def convolve_fft(array, kernel, boundary='fill', fill_value=0.,
                  nan_treatment='interpolate', normalize_kernel=True,
                  normalization_zero_tol=1e-8,
                  preserve_nan=False, mask=None, crop=True, return_fft=False,
-                 fft_pad=None, psf_pad=None, quiet=False,
```
Comment (bsipocz) [inline astropy/convolution/convolve.py]: `quiet` is not used for anything, so I've removed it. Maybe it needs to be mentioned in the changelog, too?

File: astropy/convolution/convolve.py [hunk 2]
```diff
@@ -358,10 +358,10 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
             new_result = Kernel1D(array=result)
         elif isinstance(passed_array, Kernel2D):
             new_result = Kernel2D(array=result)
+        else:
+            raise TypeError("Only 1D and 2D Kernels are supported.")
         new_result._is_bool = False
         new_result._separable = passed_array._separable
-        if isinstance(passed_kernel, Kernel):
-            new_result._separable = new_result._separable and passed_kernel._separable
```
Comment (keflavich) [inline astropy/convolution/convolve.py]: This looks like a potentially important functional change - is it intentional?

File: astropy/convolution/convolve.py (hunk 1 above)
Comment (keflavich) [reply astropy/convolution/convolve.py]: Since it wasn't doing anything... maybe not?

File: astropy/convolution/convolve.py [hunk 3]
```diff
@@ -672,7 +665,7 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0.,
             # add the shape lists (max of a list of length 4) (smaller)
             # also makes the shapes square
             fsize = 2 ** np.ceil(np.log2(np.max(arrayshape + kernshape)))
-        newshape = np.array([fsize for ii in range(array.ndim)], dtype=int)
+        newshape = np.array([fsize for _ in range(array.ndim)], dtype=int)
```
Comment (keflavich) [inline astropy/convolution/convolve.py]: Is this a widely-used convention?  I don't have a problem with it, just curious

Comment (pllim) [reply astropy/convolution/convolve.py]: Usually when there is a loop like this, the loop itself is unnecessary in most cases. Why not do something like this instead?
```python
newshape = np.full((array.ndim, ), fsize, dtype=int)
```

File: astropy/convolution/convolve.py (hunk 2 above)
Comment (bsipocz) [reply astropy/convolution/convolve.py]: `new_result._separable` has already been set to ``passed_array._separable`` the line above. I don't see how this line could change it.

Comment (keflavich) [reply astropy/convolution/convolve.py]: This line is checking whether the `passed_kernel` is _also_ separable; it's different from `passed_array`.  That can be different, no?

File: astropy/convolution/convolve.py (hunk 3 above)
Comment (bsipocz) [reply astropy/convolution/convolve.py]: yeah, my ide was complaining with ``ii`` not being used, but I didn't spend any brain power to something better like @pllim suggests. I've seen `_` being used for unnecessary variables several places, either in loops or in unpackings, but I'm not sure how much "official" it is.

**Comment by bsipocz** [comment]
Somehow I cannot respond to the 2nd inline comment about the removal of `quiet`. The only issue could be if someone was using it as a positional argument that now disappears.
Btw, I'm adding the "use kwarg only arguments" to the discussion list for the meeting,  I think this function (among many others) could benefit by it.

File: astropy/convolution/convolve.py (hunk 1 above)
Comment (bsipocz) [reply astropy/convolution/convolve.py]: The only issue could be if someone was using it as a positional argument that now disappears.
Btw, I'm adding the "use kwarg only arguments" to the discussion list for the meeting, I think this function (among many others) could benefit by it.

File: astropy/convolution/convolve.py (hunk 2 above)
Comment (bsipocz) [reply astropy/convolution/convolve.py]: oh, sure. I can't read apparently (did double check it so many times, but always read passed_kernel as passed_array)...

Comment (keflavich) [reply astropy/convolution/convolve.py]: :+1: that's what code review is for

File: astropy/convolution/convolve.py [hunk 4]
```diff
@@ -304,50 +306,48 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
                 array_to_convolve[pad_width[0]:array_shape[0]+pad_width[0]] = array_internal
             elif array_internal.ndim == 2:
                 array_to_convolve[pad_width[0]:array_shape[0]+pad_width[0],
-                             pad_width[1]:array_shape[1]+pad_width[1]] = array_internal
-            elif array_internal.ndim == 3:
+                                  pad_width[1]:array_shape[1]+pad_width[1]] = array_internal
+            else:
```
Comment (pllim) [inline astropy/convolution/convolve.py]: Was this change discussed?

Comment (bsipocz) [reply astropy/convolution/convolve.py]: ndim cannot be >3, line 244 already throws a `NotImplementedError`. But on the other hand I cannot remember why I changed it, it shouldn't matter.

Comment (pllim) [reply astropy/convolution/convolve.py]: Okie dokie then. Thanks for the clarification!

**Comment by pllim** [comment]
Thanks, @bsipocz !

**Pull request merged**